LiteCore/Storage/SQLiteDataFile.cc

```diff
@@ -293,6 +296,33 @@ namespace litecore {
             if (key.buf == nullptr || key.size != kEncryptionKeySize[alg])
                 error::_throw(error::InvalidParameter);
         }
+        bool success = _decrypt(alg, key);
+#if __APPLE__
+        if (!success && alg == kAES256) {
+            // If using AES256, retry with AES128 for backward compatibility with earlier versions:
+            logInfo("Retrying decryption with AES128...");
+            reopenSQLiteHandle();
+            success = _decrypt(kAES128, slice(key.buf, kEncryptionKeySize[kAES128]));
+            if (success) {
```

Comment:
How do we propagate the error up if the success == false? Should we throw an exception here?

Reply:
There is supposed to be a `throw` at the end if `success` is false ... I did a bunch of editing of that function and it looks like it got lost somehow. I'll restore it.